internal/provider/vdc/vdc_group_datasource.go

```diff
@@ -0,0 +1,275 @@
+	SiteName             types.String `tfsdk:"site_id"`
+	Name                 types.String `tfsdk:"name"`
+	ID                   types.String `tfsdk:"id"`
+}
+
+var vdcModelAttrTypes = map[string]attr.Type{
+	"fault_domain_tag":       types.StringType,
+	"network_provider_scope": types.StringType,
+	"is_remote_org":          types.BoolType,
+	"status":                 types.StringType,
+	"site_id":                types.StringType,
+	"site_name":              types.StringType,
+	"name":                   types.StringType,
+	"id":                     types.StringType,
+}
+
+func (d *vdcGroupDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_" + categoryName + "_" + "group"
+}
+
+func (d *vdcGroupDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Provides a data source to read vDC group and reference in other resources.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Optional: true,
+				Computed: true,
+				Validators: []validator.String{
+					stringvalidator.ExactlyOneOf(path.MatchRoot("name")),
+				},
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: "The name of the vDC Group.",
+				Optional:            true,
+				Computed:            true,
+				Validators: []validator.String{
+					stringvalidator.ExactlyOneOf(path.MatchRoot("id")),
+				},
+			},
+			"description": schema.StringAttribute{
+				MarkdownDescription: "The description of the vDC Group.",
+				Computed:            true,
+			},
+			"error_message": schema.StringAttribute{
+				MarkdownDescription: "The name of the vDC Group.",
+				Computed:            true,
+			},
+			"dfw_enabled": schema.BoolAttribute{
+				MarkdownDescription: "Distributed firewall status.",
+				Computed:            true,
+			},
+			"local_egress": schema.BoolAttribute{
+				MarkdownDescription: "Status whether local egress is enabled for a universal router belonging to a universal VDC group.",
+				Computed:            true,
+			},
+			"network_pool_id": schema.StringAttribute{
+				MarkdownDescription: "ID of used network pool of the vDC Group.",
+				Computed:            true,
+			},
+			"network_pool_universal_id": schema.StringAttribute{
+				MarkdownDescription: "The network provider’s universal id that is backing the universal network pool.",
+				Computed:            true,
+			},
+			"network_provider_type": schema.StringAttribute{
+				MarkdownDescription: "Defines the networking provider backing the VDC Group.",
+				Computed:            true,
+			},
+			"status": schema.StringAttribute{
+				MarkdownDescription: "The status that the group can be in 'SAVING', 'SAVED', 'CONFIGURING'," +
+					" 'REALIZED', 'REALIZATION_FAILED', 'DELETING', 'DELETE_FAILED', 'OBJECT_NOT_FOUND'," +
+					" 'UNCONFIGURED'",
```

Comment:
The list is present 2 times. I suggest to create `const`

```suggestion
				MarkdownDescription: "The status that the group can be in `SAVING`, `SAVED`, `CONFIGURING`," +
					" `REALIZED`, `REALIZATION_FAILED`, `DELETING`, `DELETE_FAILED`, `OBJECT_NOT_FOUND`," +
					" `UNCONFIGURED`",
```